source/if-any.d.ts

```diff
@@ -0,0 +1,29 @@
+import type {IsAny} from './is-any';
+
+/**
+An `If`/`Else`-like type that resolves depending on whether the given type is `any`.
+
+If the given type `T` is `any`, the returned type is `TypeIfAny`. Otherwise, the return type is `TypeIfNotAny`. If only `T` is specified, `TypeIfAny` will be `true` and `TypeIfNotAny` will be false.
```

Comment:
This description is confusing as you don't immediately know what `TypeIfAny` is. I suggest dropping it in favor of a good example.

For example, this is much clearer:

```js
type ShouldBeNever = IfAny<value, 'is any', 'is not any'>;
```